code/game/objects/items/binoculars.dm

```diff
@@ -161,25 +170,32 @@
 	to_chat(user, "<span class='notice'>INITIATING LASER TARGETING. Stand still.</span>")
 	if(!do_after(user, max(1.5 SECONDS, target_acquisition_delay - (2.5 SECONDS * user.skills.getRating("leadership"))), TRUE, TU, BUSY_ICON_GENERIC) || world.time < laser_cooldown || laser)
 		return
-	if(mode)
-		var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
-		coord = LT
-		to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(coord)
-			if(!do_after(user, 50, TRUE, coord, BUSY_ICON_GENERIC))
-				QDEL_NULL(coord)
-				break
-	else
-		to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
-		var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
-		laser = LT
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(laser)
-			if(!do_after(user, 50, TRUE, laser, BUSY_ICON_GENERIC))
-				QDEL_NULL(laser)
-				break
-
+	switch(mode)
+		if(1)
+			var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
+			coord = LT
+			to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(coord)
+				if(!do_after(user, 50, TRUE, coord, BUSY_ICON_GENERIC))
+					QDEL_NULL(coord)
+					break
+		if(0)
+			to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
+			var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
+			laser = LT
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(laser)
+				if(!do_after(user, 50, TRUE, laser, BUSY_ICON_GENERIC))
+					QDEL_NULL(laser)
+					break
+		if(2)
+			if((GLOB.marine_main_ship?.rail_gun?.last_firing + 600) > world.time)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The Rail Gun hasn't cooled down yet!</span>")
+			else if(!targ_area)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>No target detected!</span>")
+			else
+				GLOB.marine_main_ship?.rail_gun?.fire_rail_gun(TU,usr)
```

Comment:
Any reason the 3rd mode doesn't have a `do_after`?

Reply:
Direct copy and paste from the command console fire button.

Reply:
Because this is an on the field type thing, i'd prefer we had something that would allow some counter play, as this can still be pretty strong.